tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -482,12 +498,63 @@ func (az *AzureProvider) constructStorageProfile(imageId string) (storageProfile
 }
 
 // getWorkerBackendPoolID gets the backend pool id of the worker node loadbalancer
-func (az *AzureProvider) getWorkerBackendPoolID() *string {
-	// We assume that BackendPoolID for worker vm follows the following format
-	// /subscriptions/$SUBSCRIPTID/resourceGroups/$INFRAID-rg/providers/Microsoft.Network/loadBalancers/$INFRAID/backendAddressPools/$INFRAID
-	var wbpi = fmt.Sprintf("/subscriptions/%s/resourceGroups/%[2]s-rg/providers/"+
-		"Microsoft.Network/loadBalancers/%[2]s/backendAddressPools/%[2]s", az.subscriptionID, az.infraID)
-	return &wbpi
+func (az *AzureProvider) getWorkerBackendPoolID(ctx context.Context) (*string, error) {
+	// Create a filter to get a resource of type virtualMachine which contains worker in its name
+	filter := "resourceType eq 'Microsoft.Compute/virtualMachines' and substringof('worker', name)"
+	// number of resources to retrieve
+	// we are retrieving just 1 resource as all the worker nodes have same loadbalancers, any workerVM works for our case
```

Comment:
This is not entirely true, what if the first node is behind another loadbalancer which was custom created. In general you're making assumptions which are not valid in all the cases.